client_cli/src/main.rs

```diff
@@ -55,6 +55,24 @@ impl MetadataArgs {
         Ok(value.unwrap_or_default())
     }
 }
+/// Wrapper around Value to accept possible values and fallback to json
+#[derive(Debug, Clone)]
+pub struct ValueArg(Value);
+
+impl FromStr for ValueArg {
+    type Err = Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        s.parse::<bool>()
+            .map(Value::Bool)
+            .or_else(|_| s.parse::<Ipv4Addr>().map(Value::Ipv4Addr))
+            .or_else(|_| s.parse::<Ipv6Addr>().map(Value::Ipv6Addr))
+            .or_else(|_| s.parse::<NumericValue>().map(Value::Numeric))
+            .or_else(|_| s.parse::<PublicKey>().map(Value::PublicKey))
+            .or_else(|_| serde_json::from_str::<Value>(s).map_err(|e| e.into()))
+            .map(ValueArg)
+    }
```

Comment:
Please cover with unit tests, so that we can also see how actual values look like.

Reply:
nvbnvb

Reply:
I think PRs separation makes sense when other changes are large and not very related. Here, adding unit tests are directly related to the feature you are implementing and it is a good manner (imo) to accompany new feature with its tests.